### PR TITLE
test: consolidate to improve speed and flakiness

### PR DESCRIPTION
**Pull request by jhpedemonte**
#### What do these changes do/fix?
Consolidate `dynamic_facets` test file as per Cypress best practices. Hopefully will fix test flakiness (and speed it up).

#### How do you test/verify these changes?

Run Cypress tests

### CODE REVIEW

File: examples/discovery-search-app/cypress/integration/passages/spec.ts [hunk 1]
```diff
@@ -1,116 +1,94 @@
...
+
+    // and clicking on "CI Document" for a result
+    cy.findByLabelText('Back to search').click();
+    cy.get('button[data-testid="search-result-element-preview-button"]')
+      .contains('View table in document')
+      .click();
+    cy.findByText('Content Intelligence').click();
+
+    // diplays filters under Attribute
+    cy.findByText('Select labels to filter elements').should('exist');
```
Comment (NickMele) [inline examples/discovery-search-app/cypress/integration/passages/spec.ts:64]: Are these using the testing library helpers? Very cool if so.

File: examples/discovery-search-app/cypress/integration/passages/spec.ts [hunk 2]
```diff
@@ -1,116 +1,94 @@
...
+
+    // the passage text in each result is dangerously rendered
+    cy.get('.bx--search-result')
+      .get('em')
+      .contains('IBM')
+      .should('exist');
+
+    // passages and tables can be displayed in the same result
+    cy.get('.bx--search-result')
+      .filter(':contains("This result has a passage and a table")')
```
Comment (NickMele) [inline examples/discovery-search-app/cypress/integration/passages/spec.ts:88]: If you are using the testing library helpers, are there not better ones to handle these kinds of cases?

```ts
// dont feel great about the class selector here. is there a label or something you can use?
cy.get('.bx--search-result') 
  .queryByText('This result has a passage and a table')
```

**Comment by jhpedemonte** [comment]
Built 3 times in a row successfully. Previously, one or both of these were quite flaky.

**Comment by jhpedemonte** [comment]
@NickMele I didn't change any of the actual test code, just wanted to refactor into a single large test. We can work on fixing these up sometime later.

**Pull request merged**